### PR TITLE
fix(#3399): Improve operator log level adjustments

### PR DESCRIPTION
**Pull request by christophd**
- Use zap log level enabler implementation to set global log level
- Use Camel K logger wrapper in operator
- Use Debug wrapper methods
- Add some documentation of LOG_LEVEL envvar and its setting via kamel CLI

### CODE REVIEW

File: pkg/cmd/operator/operator.go [hunk 1]
```diff
@@ -120,16 +112,12 @@ func Run(healthPort, monitoringPort int32, leaderElection bool, leaderElectionID
...
 
-	logLevelEnabler := loglvl{
-		Level: logLevel,
-	}
-
-	logf.SetLogger(zap.New(func(o *zap.Options) {
+	logf.SetLogger(zapctrl.New(func(o *zapctrl.Options) {
 		o.Development = false
-		o.Level = logLevelEnabler
+		o.Level = zap.NewAtomicLevelAt(logLevel)
```
Comment (phantomjinx) [inline pkg/cmd/operator/operator.go:117]: Am sure that this works and is better but could you just explain how. Not least might be good to stick a comment about it to give anyonelse an idea seeing as we both got confused originally.

Comment (christophd) [reply pkg/cmd/operator/operator.go:117]: I saw that zap is doing it this way when adding the default level options so I thought it must be the right way to do it 😄

Comment (christophd) [reply pkg/cmd/operator/operator.go:121]: Have added some comments

Comment (christophd) [reply pkg/cmd/operator/operator.go:121]: @phantomjinx can you have another look, please

**Review by squakez — APPROVED** [review]

**Pull request merged**